Libraries/Network/XMLHttpRequest.js

```diff
@@ -193,26 +193,25 @@ class XMLHttpRequest extends EventTarget(...XHR_EVENTS) {
       return this._cachedResponse;
     }
 
-    switch (this._responseType) {
+    switch (responseType) {
       case 'document':
         this._cachedResponse = null;
         break;
 
       case 'arraybuffer':
-        this._cachedResponse = toArrayBuffer(
-          this._responseText, this.getResponseHeader('content-type') || '');
+        this._cachedResponse = base64.toByteArray(this._response).buffer;
```

Comment:
Could you provide a bit more information on what changed here? We previously sent over the text as a string hoping that the JS could decode correctly as an array buffer, and now we send it as base64 (less efficient over the wire)?